packages/rest/src/rest.server.ts

```diff
@@ -127,7 +131,7 @@ export class RestServer extends Context implements Server, HttpServerLike {
   protected _httpServer: HttpServer | undefined;
 
   protected _expressApp: express.Application;
-  protected _routerForStaticAssets: express.Router;
+  protected _routerForStaticAssets: ExpressRouter;
```

Comment:
Let's move the implementation details of dealing with static assets down to `StaticAssetsRoute` please. 

Ideally, I'd like to see `RestServer#static` implemented as a thin wrapper delegating the registration of static-asset middleware to StaticAssetsRoute and/or our `HttpHandler` class.

See also my earlier https://github.com/strongloop/loopback-next/pull/1848#discussion_r225476644

Reply:
@bajtos Calling `app.static` after the server has started should still set up the static assets route successfully. I tried to understand what you are suggesting and moved some pieces around, but resulted in `app.static` losing functionality after the app started.

Can you elaborate more on your suggestion, please? What does the outline of `RestServer#static` look like?

Reply:
Let's wait until you have addressed the other comments and then I can try to shuffle the code around myself, I think it will be the fastest option.

Reply:
I had addressed all others, except this one. Please refresh to see the changes.